### PR TITLE
Add ability to update Github labels

### PR DESCRIPTION
**Pull request by halkeye**
# Description 
* Add ghlabels.update
* extend ghobject for standardness (might be breaking change? but since its extend i suspect its fine)
* Change getURL signature to match ghobject

- [X] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [X] Add JavaDocs and other comments
- [X] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [X] Run `mvn -P ci install site ` locally. This may reformat your code, commit those changes. If this command doesn't succeed, your change will not pass CI.

### CODE REVIEW

**Comment by halkeye** [comment]
I didn't. I can add set name.

I think they should have fetchinto and the tests still right?

**Comment by halkeye** [comment]
https://github.com/github-api/github-api/blob/6bc617c488e96ea4e8040b54754f13ec68356bbe/src/test/java/org/kohsuke/github/AppTest.java#L814-L816

Did i accidentally fix this?

File: src/main/java/org/kohsuke/github/GHLabel.java [hunk 1]
```diff
@@ -13,7 +15,7 @@
  * @see GHIssue#getLabels() GHIssue#getLabels()
  * @see GHRepository#listLabels() GHRepository#listLabels()
  */
-public class GHLabel {
+public class GHLabel extends GHObject {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHLabel.java]: Does this actually have all the fields from GHObject? I don't think it does.

**Comment by bitwiseman** [comment]
@halkeye 
Sorry for delay, I was on vacation. 

> Did i accidentally fix this?

Yes, you kind of did - by using `fetchInto` instead of `send`.  But in doing so you've also changed the behavior of the class overall. 

If someone is depending on GHLabel instances to be immutable, they will get a surprise after this change.  I'm not saying it is wrong, it is just unclear what the right behavior for the library overall should be.  See #589. 

Look I'm not trying to make your life hard, I'm glad to have more contributors. I'm happy to make time to chat in real-time, so we can get this resolved and merged.

**Comment by bitwiseman** [comment]
@halkeye 
You around?
I'm going to go ahead cut the next release. I'm be happy to do another one when we hammer this out as well.

**Comment by bitwiseman** [comment]
@halkeye  Please take a look at #724 .

**Pull request closed without merging**